### PR TITLE
fix(bazel): Get versions from latest-versions.ts & use semver check

### PR DESCRIPTION
**Pull request by kyliau**
When `@angular/bazel` is installed, a postinstall script is run to make sure that
the npm version is *exactly* the same as the Angular repository installed by
Bazel. This check is overly stringent. Instead, it should enforce that the
version satisfies the range check instead. This check is consistent with the range
defined in `angular-cli/packages/schematics/angular/utility/latest-versions.ts`.

This commit also fixes the Bazel workspace to use the same Rxjs version if it's
already installed.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [x] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/bazel/check_version.js [hunk 1]
```diff
@@ -56,14 +57,9 @@ if (isBazelManagedDeps()) {
...
-    // TODO: we might need to support a range here.
-    // For example, if you end up with @angular/bazel@6.1.8 and
-    // @angular/bazel@6.1.9 both installed one of the postinstalls is
-    // guaranteed to fail since there's only one version of
-    // angular
-    throw new Error(`Expected angular repository to be version ${
-        npmPackageVersion} but found ${angularPackageVersion}`);
+  const range = `~${angularPackageVersion}`;
+  if (!semver.satisfies(npmPackageVersion, range)) {
+    throw new Error(`Expected angular npm version ${npmPackageVersion} to satisfy ${range}`);
```
Comment (alexeagle) [inline packages/bazel/check_version.js]: could we print more info to help the user resolve the problem? what steps should they take?

File: packages/bazel/package.json [hunk 2]
```diff
@@ -18,6 +18,7 @@
     "@bazel/typescript": "^0.21.0",
     "@schematics/angular": "^7.0.4",
     "@types/node": "6.0.84",
+    "semver": "^5.6.0",
```
Comment (alexeagle) [inline packages/bazel/package.json]: should there be a yarn.lock change to go with this?

File: packages/bazel/src/schematics/bazel-workspace/index_spec.ts [hunk 3]
```diff
@@ -59,3 +60,14 @@ describe('Bazel-workspace Schematic', () => {
     });
   });
 });
+
+describe('clean', () => {
+  it('should extract version string', () => {
+    [['1.2.3', '1.2.3'], ['  1.2.3', '1.2.3'], ['1.2.3  ', '1.2.3'], ['~1.2.3', '1.2.3'],
+     ['^1.2.3', '1.2.3'], ['v1.2.3', '1.2.3'], ['1.2', null], ['a.b.c', null],
+    ].forEach(([v, want]) => {
```
Comment (alexeagle) [inline packages/bazel/src/schematics/bazel-workspace/index_spec.ts]: nit: it's more idiomatic to generate `it` blocks, like

```
for (const test in [['1.2.3', '1.2.3']]) {
  const [v, want] = test;
  it(`should match ${v} with ${want}`, () => {
    // assertions
  });
}
```

**Review by alexeagle — APPROVED** [review]

**Review by gregmagolan — APPROVED** [review]
LGTM

**Comment by alxhub** [comment]
@kyliau this failed to cherry-pick, @alexeagle told me to merge it to master only and someone would create a patch. Please check with him.

**Pull request closed without merging**